strawberryfields/parameters.py

```diff
@@ -1,64 +1,78 @@
-  the Operation and the RegRefs it acts on, which is appended to the Engine command queue.
+* The Operation instance is applied using its :meth:`~ops.Operation.__or__`
+  method inside an :class:`.Program` context.
+  This creates a :class:`.Command` instance that wraps
+  the Operation and the RegRefs it acts on, which is appended to the Program command queue.
 
-* Once the entire program is inputted, Engine optimizes it. This involves merging and commuting Commands
-  inside the circuit graph, the building of which requires knowledge of their dependencies, both direct and Parameter-based.
+* Once the entire program is inputted, it is compiled and optimized. This involves
+  merging and commuting Commands inside the circuit graph. The circuit graph is built
+  using the knowledge of which subsystems the Commands act and depend on.
 
-* Merging two :class:`~ops.Gate` instances of the same subclass involves
+* Merging two :class:`.Gate` instances of the same subclass involves
   adding their first parameters after equality-comparing the others. This is easily done if
-  all the parameters have an immediate value. RegRefTransforms and TensorFlow objects are more complicated,
-  but could in principle be handled.
-
-For now, we simply don't do the merge if RegRefTransforms or TensorFlow objects are involved.
+  all the parameters have an immediate numerical value.
+  RegRefTransforms and TensorFlow objects are more complicated, but could in principle be handled.
+  For now, we simply don't do the merge if RegRefTransforms or TensorFlow objects are involved.
 
-* The optimized command queue is run by Engine, which calls the :func:`~ops.Operation.apply` method
-  of each Operation in turn (and tries :func:`~ops.Operation.decompose`
+* The compiled Program is run by :class:`.Engine`, which calls the :meth:`~ops.Operation.apply` method
+  of each Operation in turn (and tries :meth:`~ops.Operation.decompose`
   if a :py:exc:`NotImplementedError` exception is raised).
 
-* :func:`~ops.ParOperation.apply` evaluates the numeric value of any
-  RegRefTransform-based Parameters using :func:`Parameter.evaluate` (other types of Parameters are simply passed through).
-  The parameter values and the subsystem indices are passed to :func:`~ops.Operation._apply`.
+* :meth:`Operation.apply` evaluates the numeric value of any
+  RegRefTransform-based Parameters using :meth:`Parameter.evaluate` (other types of Parameters are simply passed through).
+  The parameter values and the subsystem indices are passed to :meth:`Operation._apply`.
 
-* :func:`~ops.Operation._apply` "unwraps" the Parameter instances. There are three different cases:
+* :meth:`~ops.Operation._apply` "unwraps" the Parameter instances. There are three different cases:
 
   1. We still need to do some arithmetic, unwrap after it is done using p.x.
   2. No arithmetic required, use :func:`~parameters._unwrap`.
   3. No parameters are used, do nothing.
```

Comment:
1. What is the difference between 'arithmetic required' and 'no arithmetic required'? In almost all cases, I have simply used `p.x`.

2. Let's mark to change this to `p.val`, as per @co9olguy's suggestion. It is more descriptive than `p.x`.

Reply:
See `Dgate._apply()` for an example of arithmetic required. I guess the `_unwrap` function is a bit unnecessary since in most cases you only have a single parameter to unwrap.